src/bl_support.cpp

```diff
@@ -0,0 +1,157 @@
+/*
+ * Scene Connect Bootloader
+ * Copyright (C) 2019 Scene Connect 
+ * https://www.scene.community/
+ *
+ * This code is developed based on the Okra Bootloader from Okra Solar.
+ * 
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ *
+ */
+
+#include "bl_support.h"
+#include "stm32l0xx.h"
+#include "thingset_serial.h"
```

Comment:
Don't think the `thingset_serial.h` is needed here. For the normal serial `#include "mbed.h"`. But also that shouldn't be needed here. If you want to print anything, use the new `print_info` or similar functions and `#include "debug.h"`.

Reply:
Agreed and updated.